### PR TITLE
[`tests`] add CI training tests

### PR DESCRIPTION
**Pull request by younesbelkada**
This PR adds CI training tests, I faced a strange issue where the gradients were `None` for the Lora model when using `PeftModelForSequenceClassification`. This test is just to make sure that the classic `PeftModelForCausalLM` has non-None gradients as expected. 

Related failing test: https://github.com/lvwerra/trl/actions/runs/4699065621/jobs/8332136001?pr=303

Let's add other tests in a folow up PR

cc @pacman100 

Note to myself, it seems that `model.zero_grad` had a different behavior depending on `torch` version with respect to peft models

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada for adding training loop tests, LGTM! 🤗

Regarding None grads, could we get a quick example to reproduce the behaviour in order to fix it at the earliest.

**Comment by younesbelkada** [comment]
Sure, I can work on that!
